### DISCUSSION TITLE
Rename helpers to not collide with Remix

### DISCUSSION THREAD

**Discussion by WesleyKapow** [sergiodxa/remix-utils#81] [Ideas]
IDEs can auto add import lines for you. This means if you use `useLoaderData` it can auto import from remix or from remix-utils. This creates unexpected behavior as developers will often assume they are importing the hook from remix.

Proposal: Rename hooks in remix-utils to not collide with those provided by remix!

**Comment by WesleyKapow** [comment]
Just hit this again! I accidentally got the wrong `json` helper and was wondering why I now needed to `JSON.parse()` my errors.

For now, my team has opted to remove `remix-utils` to avoid these problems.

**Comment by sergiodxa** [comment]
This is actually intentional, so you can replace all useLoaderData/useActionData/json from Remix with the Remix Utils version and keep it the same way.

**Reply by arnaudambro** [reply]
hello !

I actually strongly agree with @WesleyKapow : your lib of utils is good but maybe we don't want to use them all.
I use `ClientOnly`, `getClientLocales` and `getClientIPAddress`, but not the other ones. Great utils.

But I want to keep using `useLoaderData` from `remix`, and I made a lot of mistakes already, breaking my code.

I don't say I want to leave your lib, but I'm very annoyed, and I see I'm not the only one (we're two 😁).

Two solutions 
-> rename the utilities that you names exactly like remix, to have your own variable - this would be much more logical : your library is bringing so many useful and different tools that you can't expect us to be "forced" to also use `useLoaderData` if we don't want to. I will also remember it as a lesson to external libraries ! It's like you bring your own `useState`... it doesn't feel right. Anyway, this is not a quick win for you because it's a breaking change that might cause problems to existing codebases.
-> quick win if possible : change the auto suggestion so that `remix-utils` comes after `remix`, at least. I tried to see how to do so but I couldn't find out !
<img width="565" alt="image" src="https://user-images.githubusercontent.com/31724752/179477566-a31c5caa-3c5d-4b62-9fb4-38280ce97287.png">

**Comment by sergiodxa** [comment]
Closing it since those utils has not been part of the package for a while.
